src/SLCore.UnitTests/SLCoreInstanceHandleTests.cs

```diff
@@ -477,7 +483,42 @@ private void SetUpThreadHandling(IThreadHandling threadHandling)
             compatibleNodeLocator,
             activeSolutionBoundTracker,
             configScopeUpdater,
-            threadHandling);
+            threadHandling, 
+            Substitute.For<ISLCoreRuleSettings>());
+    }
+
+    private SLCoreInstanceHandle CreateTestSubject(out ISLCoreRpcFactory slCoreRpcFactory,
```

Comment:
this feels like a copy-paste just to add 1 parameter, why?

Reply:
Otherwise, I would have to update 9 unit tests just because one more parameter was added, when those unit tests do not care at all about the rules settings. That's why this coding style with out parameters is so ugly.

But yes, doing that leads to duplicate code, so I have to find a better way.

Reply:
Having this out-parameter pattern is not ideal, so if we have more time I would refactor this, but as it is now, it's better to just add the new parameter to the existing method and update the usages.

Reply:
Simplified the discussion, by removing both overloads and using TestInitialize. It was a quick refactoring.